src/components/device/ha-device-picker.ts

```diff
@@ -322,6 +331,14 @@ export class HaDevicePicker extends SubscribeMixin(LitElement) {
     return this.value || "";
   }
 
+  private _filterChanged(ev: CustomEvent): void {
+    const filterString = ev.detail.value.toLowerCase();
+    (this.comboBox as any).filteredItems = fuzzyFilterSort<ScorableDevice>(
```

Comment:
```suggestion
    ev.target.filteredItems = fuzzyFilterSort<ScorableDevice>(
```

Reply:
Do you typically attach custom data to standard HTML elements? I figured that since we already have a device picker class, it was a bit cleaner to store the props there.